data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostBlock.java

```diff
@@ -101,24 +119,69 @@ public void handleRequest(final RestApiRequest request) throws JsonProcessingExc
       return;
     }
 
-    final SignedBeaconBlock signedBeaconBlock = request.getRequestBody();
-
-    request.respondAsync(
-        validatorDataProvider
-            .submitSignedBlock(signedBeaconBlock)
-            .thenApply(
-                result -> {
-                  if (result.getRejectionReason().isEmpty()) {
-                    return AsyncApiResponse.respondWithCode(SC_OK);
-                  } else if (result
-                      .getRejectionReason()
-                      .get()
-                      .equals(FailureReason.INTERNAL_ERROR.name())) {
-                    return AsyncApiResponse.respondWithError(
-                        SC_INTERNAL_SERVER_ERROR, result.getRejectionReason().get());
-                  } else {
-                    return AsyncApiResponse.respondWithCode(SC_ACCEPTED);
-                  }
-                }));
+    Object requestBody = request.getRequestBody();
```

Comment:
If im not mistaken `request.getRequestBody()` should be able to return you the real object so we might be able to clean this up a lot by checking `request.getRequestBody() instanceof SignedBlockContents`
and just calling `request.getRequestBody` again to pass to the submits... the Object here isn't super pretty, or the casting it causes...

Reply:
Yes, it returns the real object but we can call `request.getRequestBody()` only once, hence the `Object` and casting usage

Reply:
Maybe take a look at #7094 - i think thats probably something more like what we want, where we define the type conversion and how to identify what the string is...